diff --git a/dist/server/middlewareHandler.js b/dist/server/middlewareHandler.js
index 398aeeaa862b6b019d124b29ffa5c776069381ed..d756428139a46f4c72f28aec730f2c07074f6748 100644
--- a/dist/server/middlewareHandler.js
+++ b/dist/server/middlewareHandler.js
@@ -1,13 +1,19 @@
 import { authenticateRequest } from './authenticateRequest.js';
 import { loadOptions } from './loadOptions.js';
 import { getResponseClerkState } from './utils/index.js';
+import { getContext } from 'vinxi/http';
 
 // src/server/middlewareHandler.ts
 function createClerkHandler(eventHandler, clerkOptions = {}) {
   return (cb) => {
     return eventHandler(async ({ request, router, responseHeaders }) => {
       try {
-        const loadedOptions = loadOptions(request, clerkOptions);
+        const cloudflareCtx = getContext('cloudflare');
+        const loadedOptions = loadOptions(request, {
+          ...clerkOptions,
+          secretKey: cloudflareCtx?.env?.CLERK_SECRET_KEY ?? clerkOptions.secretKey,
+          publishableKey: cloudflareCtx?.env?.CLERK_PUBLISHABLE_KEY ?? clerkOptions.publishableKey,
+        });
         const requestState = await authenticateRequest(request, loadedOptions);
         const clerkInitialState = getResponseClerkState(requestState, loadedOptions);
         router.update({
